core/lib/object_store/src/objects.rs

```diff
@@ -63,6 +65,30 @@ macro_rules! serialize_using_bincode {
     };
 }
 
+/// Derives [`StoredObject::serialize()`] and [`StoredObject::deserialize()`] using
+/// the `json` (de)serializer. Should be used in `impl StoredObject` blocks.
+
+impl StoredObject for StorageLogsSnapshot {
+    const BUCKET: Bucket = Bucket::StorageSnapshot;
+    type Key<'a> = StorageLogsSnapshotKey;
+
+    fn encode_key(key: Self::Key<'_>) -> String {
+        format!(
+            "storage_logs_{}_part{}.json",
```

Comment:
Nit: L1 batch number and part should be left-padded with zeros to guarantee natural ordering of keys.

Reply:
I changed naming to 
`format!(
            "snapshot_l1_batch_{}_part_{:0<3}.json",
            key.l1_batch_number, key.chunk_id
        )`
in the newest version both storage logs and factory deps can be in the same file, though currently only storage_logs are chunked a I assumed there are orders of magnitudes less factory deps than there are storage logs